juju/juju
# [JUJU-2962] Update cmd repo to v3.0.5

Update the cmd repo to v3.0.5 to enable the `split` option in the `juju documentation` command to generate multiple files for the commands documentation.

Notice that this version of the `cmd` library has two new fields `seealso` and `examples` to decouple both entries from the generic documentation string so we can have a better reporting solution. This will be addressed in other PRs.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

~- [ ] Code style: imports ordered, good names, simple structure, etc~
~- [ ] Comments saying why design decisions were made~
~- [ ] Go unit tests, with comments saying what you're testing~
~- [ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing~
~- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Run:
```sh
make go-build
mkdir /tmp/docs
_build/linux_amd64/bin/juju documentation --split --out=/tmp/docs --url=https://juju.is/docs/olm
```
Check files are there:
```sh
ls /tmp/docs | wc -l
194
```
Check content:
```sh
cat /tmp/docs/add-user.md 

## Summary
Adds a Juju user to a controller.

## Usage
```<user name> [<display name>]```

## Description

The user's details are stored within the controller and will be removed when
the controller is destroyed.

A user unique registration string will be printed. This registration string 
must be used by the newly added user as supplied to complete the registration
process.

Some machine providers will require the user to be in possession of certain
credentials in order to create a model.

Examples:
    juju add-user bob
    juju add-user --controller mycontroller bob

See also:
    register
    grant
    users
    show-user
    disable-user
    enable-user
    change-user-password
    remove-user

## Options
| Flag | Default | Usage |
| --- | --- | --- |
| `--B`, `--no-browser-login` | false | Do not use web browser for authentication |
| `--c`, `--controller` |  | Controller to operate in |

---
```

Check that the url is already set at the `index.md` file:
```sh
head /tmp/docs/index.md 
# Index
0. [actions](https://juju.is/docs/olm/actions)
1. [add-cloud](https://juju.is/docs/olm/add-cloud)
2. [add-credential](https://juju.is/docs/olm/add-credential)
3. [add-k8s](https://juju.is/docs/olm/add-k8s)
4. [add-machine](https://juju.is/docs/olm/add-machine)
5. [add-model](https://juju.is/docs/olm/add-model)
6. [add-secret-backend](https://juju.is/docs/olm/add-secret-backend)
7. [add-space](https://juju.is/docs/olm/add-space)
8. [add-ssh-key](https://juju.is/docs/olm/add-ssh-key)
```
## Documentation changes

NA

## Bug reference

NA
